python/ray/serve/_private/proxy.py

```diff
@@ -668,8 +673,12 @@ async def unary_stream(
                 else:
                     yield message
 
-            context.set_code(status.code)
-            context.set_details(status.message)
+            # Only the latest code and details will take effect. If the user already
+            # set them in the context, skip setting them with Serve's default values.
+            if not context.code():
+                context.set_code(status.code)
+            if not context.details():
+                context.set_details(status.message)
```

Comment:
Same suggestion about making the conditionals more explicit.

Reply:
same as above